### PR TITLE
Drop support for GHC 9.0

### PR DESCRIPTION
**Pull request by michaelpj**
Fixes https://github.com/haskell/haskell-language-server/issues/3872.

I removed the macros automatically, so let's see if this builds!

### CODE REVIEW

File: ghcide/src/Development/IDE/Core/Rules.hs [hunk 1]
```diff
@@ -1219,12 +1215,7 @@ uses_th_qq (ms_hspp_opts -> dflags) =
 -- Depends on whether it uses unboxed tuples or sums
 computeLinkableTypeForDynFlags :: DynFlags -> LinkableType
 computeLinkableTypeForDynFlags d
-#if defined(GHC_PATCHED_UNBOXED_BYTECODE) || MIN_VERSION_ghc(9,2,0)
```
Comment (wz1000) [inline ghcide/src/Development/IDE/Core/Rules.hs:1222]: Remove this flag from the cabal file?

File: ghcide/src/Development/IDE/GHC/Compat/Core.hs [hunk 2]
```diff
@@ -703,45 +652,27 @@ instance HasSrcSpan (SrcLoc.GenLocated (SrcSpanAnn' ann) a) where
...
-#if !MIN_VERSION_ghc(9,2,0)
-splitForAllTyCoVars :: Type -> ([TyCoVar], Type)
-splitForAllTyCoVars =
-  splitForAllTys
-#endif
 
 tcSplitForAllTyVars :: Type -> ([TyVar], Type)
 tcSplitForAllTyVars =
-#if MIN_VERSION_ghc(9,2,0)
   TcType.tcSplitForAllTyVars
```
Comment (fendor) [inline ghcide/src/Development/IDE/GHC/Compat/Core.hs:664]: Should we perhaps delete this simple function definition?

Its initial purpose is gone.
This raises the question again, whether we can carve out an GHC API out of our usage...

**Review by fendor — APPROVED** [review]
LGTM, once CI is green, this looks ready to go to me.

File: ghcide/src/Development/IDE/GHC/Compat/Core.hs (hunk 2 above)
Comment (michaelpj) [reply ghcide/src/Development/IDE/GHC/Compat/Core.hs:664]: Yes, I didn't go through doing simplification, which I should...

**Comment by michaelpj** [comment]
Okay, tests hopefully fixed and I did a bit of simplification in ghcide.

**Pull request merged**